### PR TITLE
update titles to sentence case, style guide

### PR DESCRIPTION
**Pull request by im2nguyen**
### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] not a security concern

### CODE REVIEW

File: website/content/docs/connect/config-entries/ingress-gateway.mdx [hunk 1]
```diff
@@ -5,7 +5,7 @@ description: >-
   The ingress gateway configuration entry kind defines behavior to secure incoming communication between the service mesh and external sources. Use the reference guide to learn about `""ingress-gateway""` config entry parameters and exposing TCP and HTTP listeners.
 ---
 
-# Ingress Gateway Configuration Entry
+# Ingress Gateway configuration entry
```
Comment (trujillo-adam) [inline website/content/docs/connect/config-entries/ingress-gateway.mdx:8]:
```suggestion
# Ingress gateway configuration entry
```

File: website/content/docs/connect/config-entries/proxy-defaults.mdx [hunk 2]
```diff
@@ -5,7 +5,7 @@ description: >-
   The proxy defaults configuration entry kind defines default behaviors for proxies in the service mesh. Use the reference guide to learn about `""proxy-defaults""` config entry parameters.
 ---
 
-# Proxy Defaults Configuration Entry
+# Proxy Defaults configuration entry
```
Comment (trujillo-adam) [inline website/content/docs/connect/config-entries/proxy-defaults.mdx:8]:
```suggestion
# Proxy defaults configuration entry
```

File: website/content/docs/connect/config-entries/service-defaults.mdx [hunk 3]
```diff
@@ -5,7 +5,8 @@ description: >-
   The service defaults configuration entry kind defines sets of default configurations that apply to all services in the mesh. Use the examples learn how to define a default protocol, default upstream configuration, and default terminating gateway.
 ---
 
-# Service Defaults Configuration Entry
+# Service Defaults configuration entry
```
Comment (trujillo-adam) [inline website/content/docs/connect/config-entries/service-defaults.mdx:8]:
```suggestion
# Service defaults configuration entry
```

File: website/content/docs/connect/config-entries/service-intentions.mdx [hunk 4]
```diff
@@ -5,7 +5,7 @@ description: >-
   The service intentions configuration entry kind defines the communication permissions between service types. Use the reference guide to learn about `""service-intentions""` config entry parameters and how to authorize L4 and L7 communication int he service mesh with intentions.
 ---
 
-# Service Intentions Configuration Entry ((#service-intentions))
+# Service Intentions configuration entry ((#service-intentions))
```
Comment (trujillo-adam) [inline website/content/docs/connect/config-entries/service-intentions.mdx:8]:
```suggestion
# Service intentions configuration entry ((#service-intentions))
```

File: website/content/docs/connect/config-entries/service-resolver.mdx [hunk 5]
```diff
@@ -5,7 +5,7 @@ description: >-
   The service resolver configuration entry kind defines subsets of service instances that satisfy upstream discovery requests. Use the reference guide to learn about `""service-resolver""` config entry parameters and how filtering by service subsets helps route traffic based on properties like version number.
 ---
 
-# Service Resolver Configuration Entry
+# Service Resolver configuration entry
```
Comment (trujillo-adam) [inline website/content/docs/connect/config-entries/service-resolver.mdx:8]:
```suggestion
# Service resolver configuration entry
```

File: website/content/docs/connect/config-entries/service-router.mdx [hunk 6]
```diff
@@ -5,7 +5,7 @@ description: >-
   The service router configuration entry kind defines where the service mesh routes requests based on L7 network information such as header or path. Use the reference guide to learn about `""service-router""` config entry parameters and how behaviors like request timeouts, retry behavior, header modification, and path rewriting can be applied to a request based on its header or path information.
 ---
 
-# Service Router Configuration Entry
+# Service Router configuration entry
```
Comment (trujillo-adam) [inline website/content/docs/connect/config-entries/service-router.mdx:8]:
```suggestion
# Service router configuration entry
```

File: website/content/docs/connect/config-entries/service-splitter.mdx [hunk 7]
```diff
@@ -7,7 +7,7 @@ description: >-
   reference, configuration model, a complete example, and example code by use case.
 ---
 
-# Service Splitter Configuration Reference
+# Service Splitter configuration entry
```
Comment (trujillo-adam) [inline website/content/docs/connect/config-entries/service-splitter.mdx:10]:
```suggestion
# Service splitter configuration entry
```

File: website/content/docs/connect/config-entries/terminating-gateway.mdx [hunk 8]
```diff
@@ -5,7 +5,7 @@ description: >-
   The terminating gateway configuration entry kind defines behavior to secure outgoing communication between the service mesh and non-mesh services. Use the reference guide to learn about `""terminating-gateway""` config entry parameters and connecting from your service mesh to external or non-mesh services registered with Consul.
 ---
 
-# Terminating Gateway Configuration Entry
+# Terminating Gateway configuration entry
```
Comment (trujillo-adam) [inline website/content/docs/connect/config-entries/terminating-gateway.mdx:8]:
```suggestion
# Terminating gateway configuration entry
```

**Review by trujillo-adam — COMMENTED** [review]
We probably want to move away from treating configuration entry names as brand names or proper nouns.

**Comment by david-yu** [comment]
Could you review the K8s ones as well, I believe this page has a title that needs to be addressed: https://developer.hashicorp.com/consul/docs/k8s/deployment-configurations/servers-outside-kubernetes

**Pull request closed without merging**